package/MDAnalysis/topology/TXYZParser.py

```diff
@@ -117,9 +120,16 @@ def parse(self, **kwargs):
         # Guessing time
         masses = guessers.guess_masses(names)
 
+        for n in names:
+             if n.capitalize() in SYMB2Z:
+                 validated_elements.append(n.capitalize())
+             else:
+                validated_elements.append('')
+
         attrs = [Atomnames(names),
                  Atomids(atomids),
                  Atomtypes(types),
+                 Elements(np.array(validated_elements, dtype=object)),
```

Comment:
I **think** it would be better to add the `elements` attribute only if all the names are valid element symbols. Otherwise, we take the risk of mistaking names for element symbols.

Reply:
I can't remember where we got on this conversation, but I actually thought we had somehow agreed that `''` was a valid "invalid element was found" fallback case?

I know there's a few places where we allow for incomplete element arrays, but raise a user warning about it?

Reply:
An empty string is indeed what we agreed upon as a placeholder value. My logic, here, is that the column we use to fill the element attribute is defined as being either the atom names or the element symbols. If we have invalid element symbols it is likely that the column is used as atom names; then, we cannot assume that the names are actual element symbols so we should consider we do not know about the elements.

The relevant part of the Tinker documentation says:
> The .xyz file is the basic Tinker Cartesian coordinates file type. It contains a title line followed by one line for each atom in the structure. Each line contains: the sequential number within the structure, **an atomic symbol or name**, X-, Y-, and Z-coordinates, the force field atom type number of the atom, and a list of the atoms connected to the current atom.

So, at least, it is how **I** understand the problem.

Reply:
Ah ok yeah that makes sense. I guess it's like a pseudo guess to work out which of the two is happening.

Reply:
I got it but as I'm taking the guessing point of view, if the name column have invalid elements then guessing behavior will take two different patterns before and after the new guesser API as follows:
old behavior: guessing masses will be done from atom names which can lead to trashy results here and there
new behavior: guessing masses will be done from atom type, which will result in completely invalid output cause atom types are represented by numeric values
I think sticking to not guess any trashy masses is a good update, but I'm afraid it will break any current default behavior